config/configmodels/receiver.go

```diff
@@ -20,6 +20,24 @@ type Receiver interface {
 	NamedEntity
 }
 
+// CustomConfigOptions defines the interfaces for configuration customization options
+// Different custom interfaces can be added like CustomConfigValidator, CustomConfigUnmarshaller, etc.
+type CustomConfigOptions interface {
```

Comment:
As mentioned in the comment, would be good to make this a `required` func to implement.